drivers/include/bq2429x.h

```diff
@@ -0,0 +1,500 @@
+ *
+ * @param[out]  dev     Device descriptor.
+ * @param[in]   params  Device parameters.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ * @return BQ2429X_ERR_GPIO on GPIO initialization failure (CE or OTG pin
+ *         failed).
+ */
+int bq2429x_init(bq2429x_t *dev, const bq2429x_params_t *params);
+
+#if IS_USED(MODULE_BQ2429X_INT) || DOXYGEN
+/**
+ * @brief   Callback function for BQ2429x interrupts.
+ */
+typedef void (* bq2429x_int_cb_t)(void *);
+
+/**
+ * @brief   Initialize interrupt support for the device.
+ *
+ * @pre @p dev != NULL && @p cb != NULL
+ *
+ * The callback @p cb is called in an ISR context, so keep in mind that heavy
+ * work shouldn't be done there.
+ *
+ * @note @ref bq2429x_init MUST have been called before!
+ *
+ * @param[in]   dev     Device descriptor.
+ * @param[in]   cb      Callback called on interrupt.
+ * @param[in]   arg     Argument to be passed when the callback is called.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_GPIO on GPIO initialization failure.
+ */
+int bq2429x_init_int(bq2429x_t *dev, bq2429x_int_cb_t cb, void *arg);
+#endif /* IS_USED(MODULE_BQ2429X_INT) || DOXYGEN */
+
+/**
+ * @brief   Get device status.
+ *
+ * @pre @p dev != NULL && @p status != NULL
+ *
+ * @param[in]   dev         Device descriptor.
+ * @param[out]  status      Pointer where device status will be stored.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C error.
+ */
+int bq2429x_get_status(const bq2429x_t *dev, bq2429x_status_t *status);
+
+/**
+ * @brief   Get device faults.
+ *
+ * @pre @p dev != NULL && @p fault != NULL
+ *
+ * @param[in]   dev         Device descriptor.
+ * @param[out]  fault       Pointer where device faults will be stored.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C error.
+ */
+int bq2429x_get_fault(const bq2429x_t *dev, bq2429x_fault_t *fault);
+
+/**
+ * @brief   Enable OTG.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev Device descriptor.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_enable_otg(const bq2429x_t *dev);
+
+/**
+ * @brief   Disable OTG.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev Device descriptor.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_disable_otg(const bq2429x_t *dev);
+
+/**
+ * @brief   Enable battery charging.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev Device descriptor.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_enable_charge(const bq2429x_t *dev);
+
+/**
+ * @brief   Disable battery charging.
+ *
+ * @pre @p dev != NULL
+ *
+ * @param[in]   dev Device descriptor.
+ *
+ * @return BQ2429X_OK on success.
+ * @return BQ2429X_ERR_I2C on I2C failure.
+ */
+int bq2429x_disable_charge(const bq2429x_t *dev);
+
+/**
+ * @brief   Set Input Voltage Limit.
```

Comment:
Is this an upper or a lower limit?

Reply:
It's a lower limit, tested with the test application by setting the VLIM to a higher voltage than the one that was supplied through VBUS until it gave a fault.

36e9c93cc4540733c481df1f733c2f5bb0031013

Reply:
E.g.: when VBUS is 4.8V and VLIM at 5 V the device stops working.